node/src/WorkQueue.ts

```diff
@@ -47,6 +48,54 @@ export class WorkQueue {
     pipeline.lpush(this.mainQueueKey, item.id);
   }
 
+  /**
+   * Add an item to the work queue in a atomic way.
+   * This creates a pipeline and executes it on the database if the item id is not already in either the main queue or the procesing queue.
+   * 
+   * This method can be used only when database is locked
+   * 
+   * @param {Redis} db The Redis Connection.
+   * @param item The item that will be executed using the method addItemToPipeline.
+   */
+
+  async addAtomicItem(db: Redis, item: Item): Promise<boolean | null> {
+      let transactionError = true;
+      let success = false;
+      do {
+        try {
+          await db.watch(this.mainQueueKey, this.processingKey);
+  
+          const pipeNew = db.pipeline();
+  
+          pipeNew.lpos(this.processingKey, item.id);
+  
+          pipeNew.lpos(this.mainQueueKey, item.id);
+  
+          const pipeResult = await pipeNew.exec();
+  
+
+          if (pipeResult && (pipeResult[0][1] !== null || pipeResult[1][1] !== null)) {
+            transactionError = false; // Item already exists in either mainQueue or processingKey
+            return false;
+          }
+  
+          const pipeEx = db.multi();
+          this.addItemToPipeline(pipeEx, item);
+          let results = await pipeEx.exec()
+          if (results && results[0][1] === "OK") {
+            success = true;
+            transactionError = false;
+            return transactionError;
+          }
+        } finally {
+          db.unwatch();
+        }
+      } while (transactionError);
+  
+      return success;
```

Comment:
The loop can never be broken. Currently it always returns after setting `transactionError = false`. It's just a forever loop. No need for the return statement afterwards!

Also, there's no need for the `success` or `transactionError` variables. Just remove them.

You can write a loop that goes forever using
```javascript
for (;;) {
    // This will loop forever
}
```